sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/ConsumePartitionEventsAsyncRetrySample.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.messaging.eventhubs.models.EventPosition;
+import com.azure.messaging.eventhubs.models.LastEnqueuedEventProperties;
+import com.azure.messaging.eventhubs.models.PartitionEvent;
+import com.azure.messaging.eventhubs.models.ReceiveOptions;
+import reactor.core.Disposable;
+import reactor.core.Disposables;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.util.retry.Retry;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+public final class ConsumePartitionEventsAsyncRetrySample {
+    public static void main(String[] args) throws Exception {
+        // "<<fully-qualified-namespace>>" will look similar to "{your-namespace}.servicebus.windows.net"
+        // "<<event-hub-name>>" will be the name of the Event Hub instance you created inside the Event Hubs namespace.
+        final String fullyQualifiedNamespace = "<<fully-qualified-namespace>>";
+        final String eventHubName = "<<event-hub-name>>";
+        //
+        // The "$Default" consumer group is created by default. This value can be found by going to the Event Hub
+        // instance you are connecting to, and selecting the "Consumer groups" page.
+        final String consumerGroup = "$Default";
+        //
+        // "<<partition-id>>" is the identifier of the partition within the Event Hub.
+        final String partitionId = "<<partition-id>>";
+        //
+        // The position to start reading events from, latest() means receive from the next event to be enqueued.
+        final EventPosition initialEventPosition = EventPosition.latest();
+        //
+        // The credential used is DefaultAzureCredential because it combines commonly used credentials
+        // in deployment and development and chooses the credential to used based on its running environment.
+        // More information can be found at: https://learn.microsoft.com/java/api/overview/azure/identity-readme
+        final TokenCredential tokenCredential = new DefaultAzureCredentialBuilder().build();
+        //
+        // The handler to process each event from the partition identified by the 'partitionId'.
+        final Consumer<PartitionEvent> eventConsumer = new Consumer<PartitionEvent>() {
```

Comment:
I would consider making these private static methods because anonymous classes are hard to follow/read.
```java
private static void onEvent(PartitionEvent partitionEvent) {
    System.out.println("Processing the event at offset: " + partitionEvent.getData().getOffset());
}
```

or
```
final Consumer<PartitionEvent> eventConsumer = (partitionEvent) -> {
    System.out.println("Processing the event at offset: " + partitionEvent.getData().getOffset());
};
```

though I prefer the first one. :)